components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -604,6 +606,25 @@ fn cast_any_as_string<T: ConvertTo<Bytes> + Evaluable + EvaluableRet>(
     }
 }
 
+#[rpn_fn(nullable, capture = [ctx, extra])]
```

Comment:
Please considering remove `nullable` here so we don't need to check this, take [this PR](https://github.com/tikv/tikv/pull/8728) as ref.

Reply:
Please fix this, rest lgtm

Reply:
According to Alex Chi, writer can't be used together with ctx, and the amoung of bytes written is limited, therefore we can keep it this way.

Reply:
Remove `nullable` and add `writer` is different thing.